mlflow/tracing/provider.py

```diff
@@ -244,15 +219,33 @@ def f():
 
     @functools.wraps(f)
     def wrapper(*args, **kwargs):
-        was_trace_enabled = _is_enabled()
-        if was_trace_enabled:
-            disable()
-            try:
+        is_func_called = False
+        try:
+            was_trace_enabled = _is_enabled()
+            if was_trace_enabled:
+                disable()
+                try:
+                    is_func_called = True
+                    return f(*args, **kwargs)
+                finally:
+                    enable()
+            else:
+                is_func_called = True
+                return f(*args, **kwargs)
+        # We should only catch the exception from disable() and enable()
+        # and let other exceptions propagate.
+        except MlflowTracingException as e:
+            _logger.warning(
+                f"An error occurred while disabling or re-enabling tracing: {e} "
+                "The original function was still executed, but the tracing "
```

Comment:
At this point, the original function may not be exeucted, right?

Reply:
True, more appropriate term should be `will be`